pkg/controller/nodeipam/ipam/cidrset/cidr_set.go

```diff
@@ -139,22 +141,21 @@ func (s *CidrSet) AllocateNext() (*net.IPNet, error) {
 	s.Lock()
 	defer s.Unlock()
 
-	nextUnused := -1
+	if s.count >= s.maxCIDRs {
+		return nil, ErrCIDRRangeNoCIDRsRemaining
+	}
+
 	for i := 0; i < s.maxCIDRs; i++ {
 		candidate := (i + s.nextCandidate) % s.maxCIDRs
 		if s.used.Bit(candidate) == 0 {
-			nextUnused = candidate
-			break
+			s.nextCandidate = (candidate + 1) % s.maxCIDRs
+			s.used.SetBit(&s.used, candidate, 1)
```

Comment:
It might be interesting to add a prometheus metric (distribution?) to track number of tries it takes to allocate

Reply:
still missing testing for the metrics, but check if the metrics added are ok